src/org/opendatakit/briefcase/export/Csv.java

```diff
@@ -77,32 +75,35 @@ static Csv repeat(FormDefinition formDefinition, Model groupModel, ExportConfigu
     );
   }
 
-  /**
-   * Returns true if the grandparent node of the given Model is the model's root
-   * <p>
-   * Example 1:
-   * <p>
-   * <code><pre>
-   * &lt;data&gt;
-   * &nbsp;&nbsp;&lt;/some_field&gt;
-   * &lt;/data&gt;
-   * </pre></code>
-   * <p>
-   * In this example:
-   * <ul>
-   * <li>&lt;data&gt; has a parent with <code>null</code> name</li>
-   * <li>Returns true on &lt;some_field&gt;</li>
-   * </ul>
-   */
-  private static boolean grandParentIsRoot(Model current) {
-    return current.hasParent()
-        && current.getParent().hasParent() // Check if current has a grandparent
-        && current.getParent().getParent().getName() == null; // The root node has a null name
+  private static Path buildMainOutputPath(FormDefinition formDefinition, ExportConfiguration configuration) {
+    return configuration.getExportDir().resolve(String.format(
+        "%s.csv",
+        configuration.getFilenameBase(formDefinition.getFormName())
+    ));
+  }
+
+  private static Path buildRepeatOutputPath(FormDefinition formDefinition, Model groupModel, ExportConfiguration configuration) {
+    return configuration.getExportDir().resolve(String.format(
+        "%s-%s.csv",
+        configuration.getFilenameBase(formDefinition.getFormName()),
+        stripIllegalChars(groupModel.getName())
+    ));
+  }
+
+  private static Path buildRepeatOutputPath(FormDefinition formDefinition, Model groupModel, ExportConfiguration configuration, int sequenceNumber) {
+    return configuration.getExportDir().resolve(String.format(
+        "%s-%s~%d.csv",
+        configuration.getFilenameBase(formDefinition.getFormName()),
+        stripIllegalChars(groupModel.getName()),
+        sequenceNumber
+    ));
```

Comment:
I’m tempted to suggest you remove the duplication in these two methods.

Reply:
Ugh! Sorry about forgetting to answer this comment!!

I've been thinking and trying things to remove dupe code here but nothing seems right. Do you have any suggestions?